addon/ShareProvider.js

```diff
@@ -210,10 +126,140 @@ ShareProvider.prototype = {
   init: Task.async(function*() {
     yield this._setupProviders();
     this._createButton();
-    Overlay.init(this.eventTracker);
+    Overlay.init(this);
   }),
 
   /**
+   * Return the installed social share providers.
+   */
+  get socialProviders() {
+    const defaultSort = ["Facebook", "Twitter", "Tumblr", "LinkedIn", "Yahoo Mail", "Gmail"];
```

Comment:
maybe store these as a global constant? if not in `common/constants` at least up top?